components/mdns/mdns.c

```diff
@@ -18,7 +18,7 @@
 #include "esp_log.h"
 #include "esp_random.h"
 
-#if CONFIG_ETH_ENABLED && CONFIG_MDNS_PREDEF_NETIF_ETH
+#if CONFIG_ETH_ENABLED
```

Comment:
suggest reverting this change

```suggestion
#if CONFIG_ETH_ENABLED && CONFIG_MDNS_PREDEF_NETIF_ETH
```

The reason is that even if we're using a custom Ethernet interface there's no need for `esp_eth` headers (as the netif itself is well abstracted), which helps reduce component coupling.